contracts/events/TopicEvent.sol

```diff
@@ -491,19 +458,43 @@ contract TopicEvent is ITopicEvent, Ownable, ReentrancyGuard {
     * @return The amount of BOT tokens won.
     */
     function calculateBotTokensWon() 
-        private
+        public
         view
+        inCollectionStatus()
         returns (uint256) 
     {
-        uint256 losingResultsTotal = 0;
-        for (uint8 i = 0; i < numOfResults; i++) {
-            if (i != finalResultIndex) {
-                losingResultsTotal = losingResultsTotal.add(balances[i].totalVoteBalance);
-            }
-        }
+        uint256 betBalance = balances[finalResultIndex].betBalances[msg.sender];
+        uint256 voteBalance = balances[finalResultIndex].voteBalances[msg.sender];
+        uint256 totalContribution = betBalance.add(voteBalance);
 
-        uint256 senderVoteBalance = balances[finalResultIndex].voteBalances[msg.sender];
-        uint256 winningResultTotal = balances[finalResultIndex].totalVoteBalance;
-        return senderVoteBalance.mul(losingResultsTotal).div(winningResultTotal).add(senderVoteBalance);
+        uint256 totalWinningBets = balances[finalResultIndex].totalBetBalance;
+        uint256 totalWinningVotes = balances[finalResultIndex].totalVoteBalance;
+        uint256 totalWinningContribution = totalWinningBets.add(totalWinningVotes);
+
+        uint256 totalLosingVotes = totalBotValue.sub(totalWinningVotes);
+        return totalContribution.mul(totalLosingVotes).div(totalWinningContribution).add(voteBalance);
```

Comment:
will there be any precision problem with `mul` & `div`? If yes, we can round it down to lower bound

Reply:
Solidity automatically deals with decimals by dropping them. So when users withdraw, there will be leftover remainder of QTUM and BOT. I was thinking we can just withdraw the remainder into our accounts.

I haven't figured out a good way to distribute the remainder (or calculate the remainder ahead of time) so that we know how much will be left. If you have any ideas, I'm all ears!